### DISCUSSION TITLE
How to get only cropped pages data

### DISCUSSION THREAD

**Discussion by kathimohan** [py-pdf/pypdf#1501] [Q&A]
Cropped a page using pypdf but when tried to read text, it is providing all the data instead of only cropped one. How to get only cropped page's data

from PyPDF2 import PdfWriter, PdfReader
from PyPDF2.generic import RectangleObject
import pdfplumber

reader = PdfReader('abc.pdf')
writer = PdfWriter()

page1  = reader.pages[1]
mb = page1.mediabox
page1.cropbox = RectangleObject((41.82533333333333, 771.0659999999999, 552.965,  741.742))
writer.add_page(page1)
#print(page1.extractText())
with open('new.pdf', 'wb') as f:
    writer.write(f)
pdf_name = "new.pdf"
with pdfplumber.open(pdf_name) as pdf:
    page = pdf.pages[0]
    tb=page.extract_text()
    print(tb)
reader = PdfReader("new.pdf")
page = reader.pages[0]
print(page.extract_text())

**Comment by pubpub-zz** [comment]
disccused in https://github.com/py-pdf/PyPDF2/issues/1508#issuecomment-1357341677 :

>the crop box is only a reframing, the whole text is still present and available : PyPDF2 is processing the whole context and therefore will extract/process all text. The best solution if you want to extract only the displayed text is to use a visitor function with the text extraction (https://pypdf2.readthedocs.io/en/latest/user/extract-text.html#using-a-visitor) validating the extraction of only the text within the cropbox.

**Reply by kathimohan** [reply]
Instead of text I need a page and need to attach it to another cropped page. Is there any way to do that?

**Reply by pubpub-zz** [reply]
@kathimohan,
your request seems to not be linked with this thread. Can you open a new one and describe more precisely your request
